src/store/models/index.ts

```diff
@@ -0,0 +1,258 @@
+}
+
+const booleanRE = /^([01])|(true)|(false)$/;
+const booleanTrue = /^(1)|(true)$/;
+
+export function parseBoolean(_value: boolean | string | number) {
+	if (typeof _value === 'boolean') return _value;
+	if (typeof _value === 'number' && (_value === 1 || _value === 0)) {
+		return _value === 1;
+	}
+	if (typeof _value === 'string' && booleanRE.test(_value)){
+		return booleanTrue.test(_value);
+	}
+	throw new BooleanParseException(`The value '${_value}' is not a boolean`);
+
+}
+
+export const user_roles = ['admin', 'instructor', 'TA', 'student'];
+
+export function parseUserRole(role: string) {
+	if (user_roles.findIndex((v) => v === role) < 0) {
+		const err = new ParseError('InvalidRole', `The value '${role}' is not a valid role`);
+		err.field = 'role';
+		throw err;
+	}
+	return role;
+}
+
+export class RequiredFieldsException extends ParseError {
+	constructor(_field: string, message: string){
+		super('RequiredFieldsException', message);
+		super.field = _field;
+	}
+}
+
+export class InvalidFieldsException extends ParseError {
+	constructor(_field: string, message: string) {
+		super('InvalidFieldsException', message);
+		super.field = _field;
+	}
+}
+
+export type generic = string|number|boolean;
+
+export type ParseableModel = Dictionary<generic|Dictionary<generic>>;
+
+export interface ModelField  {
+	[k: string]: {
+		field_type: 'string'|'boolean'|'number'|'non_neg_int'|'username'|'email'|'role';
+		default_value?: generic;
+		required?: boolean;
+	}
+}
+
+/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access,
+			@typescript-eslint/no-explicit-any */
+
+export function parseParams(_params: Dictionary<generic>,  _fields: ModelField){
+	const output_params: Dictionary<generic> = {};
+	Object.keys(_fields).forEach((key: string) => {
+		if ((_params as any)[key] == null && _fields[key].default_value !== undefined) {
+			(_params as any)[key] = _fields[key].default_value;
+		}
+		// parse each field
+		if ((_params as any)[key] != null && _fields[key].field_type === 'boolean') {
+			(output_params as any)[key] = parseBoolean((_params as any)[key]);
+		} else if ((_params as any)[key] != null && _fields[key].field_type === 'string') {
+			(output_params as any)[key] = `${(_params as any)[key] as string}`;
+		} else if ((_params as any)[key] != null && _fields[key].field_type === 'non_neg_int') {
+			(output_params as any)[key] = parseNonNegInt((_params as any)[key]);
+		} else if ((_params as any)[key] != null && _fields[key].field_type === 'username') {
+			(output_params as any)[key] = parseUsername((_params as any)[key]);
+		} else if ((_params as any)[key] != null && _fields[key].field_type === 'email') {
+			(output_params as any)[key] = parseEmail((_params as any)[key]);
+		} else if ((_params as any)[key] != null && _fields[key].field_type === 'role') {
+			(output_params as any)[key] = parseUserRole((_params as any)[key]);
+		}
+	});
+	return output_params;
+}
+// eslint-enable
```

Comment:
This doesn't work.   This needs to be 
```suggestion
/* eslint-enable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access,
	@typescript-eslint/no-explicit-any */
```

`// eslint-enable` only enables eslint if it was disabled for the file, but it does not re-enable previously disabled rules.  So if the above disabled rules are broken anywhere else in this file, eslint will continue to ignore them.

Reply:
I'll do a bigger search on this.